server/src/test/java/io/crate/execution/dml/upsert/TransportShardUpsertActionTest.java

```diff
@@ -298,4 +309,35 @@ public void testItemsWithoutSourceAreSkippedOnReplicaOperation() throws Exceptio
         verify(indexShard, times(0)).applyIndexOperationOnReplica(
             anyLong(), anyLong(), anyLong(), anyLong(), anyBoolean(), any(SourceToParse.class));
     }
+
+    @Test
+    public void test_dynamic_insert_of_integer_upcasted_to_long_can_be_replicated() throws IOException {
+        // A dynamic insert to primary creates the dynamic column, so the follow-up dynamic insert to replica is actually no longer dynamic.
+        // The follow-up dynamic insert to replica tries to resolve the DynamicReference to a SimpleReference and there is potential ClassCastException.
```

Comment:
just a suggestion for better phrasing:
```suggestion
        // The follow-up dynamic insert to replica tries to resolve the DynamicReference to a SimpleReference and there can be a potential ClassCastException.
```